server.py

```diff
@@ -117,61 +149,44 @@ def server_static(filepath):
 
 @app.get("/ranking/<code>")
 def contest_ranking(code):
-    with shelve.open(database_path) as submission_record:
-        order = [
-            (
-                user,
-                len(
-                    set(
-                        [
-                            attempt.question
-                            for attempt in submissions
-                            if (
-                                attempt.is_correct
-                                and (int(attempt.question) in contests[code].questions)
-                                and attempt.contest == code
-                                and attempt.time <= contests[code].end_time
-                                and attempt.time >= contests[code].start_time
-                            )
-                        ]
-                    )
-                ),
-            )
-            for user, submissions in submission_record.items()
-        ]
-    order.sort(key=lambda x: x[1], reverse=True)
-    order = [entry for entry in order if entry[1] > 0]
-    order = [(user, score, rank) for rank, (user, score) in enumerate(order, start=1)]
+    order = (
+        Submission.select(
+            Submission.username, fn.count(Submission.question).alias("score")
+        )
+        .where((Submission.is_correct == True) & (Submission.contest == code))
+        .group_by(Submission.username)
+        .order_by(fn.count(Submission.question).desc())
```

Comment:
umm. I'm not sure but at first look it seems this might count duplicate correct submissions as valid

Reply:
Yes. Sorry. My bad. Any idea how to resolve it?

Reply:
maybe have a score associated with a submission and only the first correct submission gets a score of 1?

Reply:
we can fix it in a later issue. Let's get the database working in this one.

Reply:
OK. database is done.

Reply:
Did not make the relationships yet though